packages/dds/map/src/directory.ts

```diff
@@ -743,29 +743,31 @@ export class SharedDirectory
 	}
 
 	/**
-	 * This checks if there is pending delete op for local delete for a subdirectory.
+	 * This checks if there is pending delete op for local delete for a any subdir in the relative path.
 	 * @param relativePath - path of sub directory.
 	 * @returns - true if there is pending delete.
 	 */
 	private isSubDirectoryDeletePending(relativePath: string): boolean {
-		const parentSubDir = this.getParentDirectory(relativePath);
-		const index = relativePath.lastIndexOf(posix.sep);
-		const dirName = relativePath.substring(index + 1);
-		return !!parentSubDir?.isSubDirectoryDeletePending(dirName);
-	}
-
-	/**
-	 * Gets the parent directory of a sub directory.
-	 * @param relativePath - path of sub directory of which parent needs to be find out.
-	 */
-	private getParentDirectory(relativePath: string): SubDirectory | undefined {
 		const absolutePath = this.makeAbsolute(relativePath);
 		if (absolutePath === posix.sep) {
-			return undefined;
+			return false;
 		}
-		const index = absolutePath.lastIndexOf(posix.sep);
-		const parentAbsPath = absolutePath.substring(0, index);
-		return this.getWorkingDirectory(parentAbsPath) as SubDirectory;
+		let currentParent = this.root;
+		const nodeList = absolutePath.split(posix.sep);
+		let start = 1;
+		while (start < nodeList.length) {
+			const subDirName = nodeList[start];
+			if (currentParent.isSubDirectoryDeletePending(subDirName)) {
+				return true;
+			}
+			const newParent = currentParent.getSubDirectory(subDirName) as SubDirectory;
```

Comment:
You could just assign to currentParent here.

Reply:
done